frontend/src/helpers/service.ts

```diff
@@ -0,0 +1,23 @@
+export type ServiceResponse = {
+    ok: boolean;
+    message: string;
+};
+
+export type ServiceError = {
+    error: string;
+    message: any;
+};
+
+export type FieldError = {
+    code: string;
+    minimum: number;
+    type: string;
+    inclusive: boolean;
+    exact: boolean;
+    message: string;
+    path: string[];
+};
+
+export function formatFieldError(errors: FieldError[]) {
```

Comment:
I am sorry for this, but I think I gonna fix question service and user service to always return a meaningful error message. So for example in case of input error, I gonna return the format that you want here. So you can just do `response.message` to show the error message. Is that okay for you?

Reply:
what about multiple input errors, will the message be pretty formatted?

Reply:
I think for each response, I will only provide one input error. For example, if there are multiple errors such as "question title too short" and "description is missing", I will only tell you that the question title is too short. And then on the second call I will then tell you hey description is missing. Is that okay? It is a bit hard to return multiple error messages at the same time and I think it is also quite hard to process it on the frontend.

Reply:
that's not very intuitive for the error feedback, can the message field return objects? something like this:
```
{
title: "title is missing"
description: "description is missing"
}
```
which makes it easier to set the error feedback for each individual field.

Reply:
I think I get what you mean, let's continue the discussion to this [PR ](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g05/issues/39)that I just created for changing the response error message.